DEVEL.rst

```diff
@@ -8,132 +8,176 @@ If something does not work as expected, please create an issue or
-sudo dnf install -y etcd redis
-sudo dnf install -y @development-tools redhat-rpm-config
-sudo dnf install -y python3 python3-virtualenv
-```
+First install the required packages:
+
+.. code-block:: shell
+
+   $ sudo dnf update -y
+   $ sudo dnf install -y etcd redis
+   $ sudo dnf install -y @development-tools redhat-rpm-config
+   $ sudo dnf install -y python3 python3-virtualenv
+
+
+Next enable the ``etcd`` and ``redis`` services:
+
+.. code-block:: shell
+
+   $ systemctl enable etcd redis
+   $ systemctl start etcd redis
 
-```
-$ systemctl enable etcd redis
-$ systemctl start etcd redis
-```
 
 
-## Pull repositories
+Pull repositories
+-----------------
+
 Commissaire consists of several projects that either provide
-common code or the actual services.
+common code or the actual services. You can clone the repos
+by following the example below:
+
+.. code-block:: shell
+
+   $ mkdir commissaire-projects
+   $ cd commissaire-projects
+   $ git clone https://github.com/projectatomic/commissaire
+   $ git clone https://github.com/projectatomic/commissaire-service
+   $ git clone https://github.com/projectatomic/commissaire-http
+   $ git clone https://github.com/projectatomic/commctl
 
-```
-mkdir commissaire-projects
-cd commissaire-projects
-git clone https://github.com/projectatomic/commissaire
-git clone https://github.com/projectatomic/commissaire-service
-git clone https://github.com/projectatomic/commissaire-http
-git clone https://github.com/projectatomic/commctl
-```
 
 After this we will install each project to setup a development
 environment.
 
 
-### Create VirtualEnv
-```
-virtualenv-3.5 devel
-. devel/bin/activate
-```
+Create VirtualEnv
+-----------------
+
+.. code-block::
+
+   $ virtualenv-3.5 devel
+   $ . devel/bin/activate
+
 
 Continue executing the following commands in the virtualenv you just created.
 
 
-## Install Commissaire
-```
-cd commissaire
-pip install -e .
-popd
-```
+Install Commissaire
+===================
 
+.. code-block::
 
-### Install Commissaire Service
-```
-cd commissaire-service
-pip install -e .
-tools/etcd_init.sh
-```
+   $ cd commissaire
+   $ pip install -e .
+   $ popd
 
-Edit your storage configuration to point to your etcd instance
-```
-cp conf/storage.conf mystorage.conf
-```
-Note: point server_url to http://127.0.0.1:2379 (not https)
+
+Install Commissaire Service
+---------------------------
+
+.. code-block::
+
+   $ cd commissaire-service
+   $ pip install -e .
+   $ tools/etcd_init.sh
+
+
+Edit your storage configuration to point to your etcd instance:
+
+.. code-block::
+
+   $ cp conf/storage.conf mystorage.conf
+
+
+**Note**: Point ``server_url`` to ``http://127.0.0.1:2379`` (not https)
 
 
 Start the service
-```
-commissaire-storage-service -c mystorage.conf &
-popd
-```
 
-### Install Commissaire Server
-```
-cd commissaire-http
-pip install -e .  # Install commissaire-http into the virtualenv
-```
+.. code-block::
+
+   $ commissaire-storage-service -c mystorage.conf &
```

Comment:
Failing here. Verified `etcd` is listening on the right port and my config is trying to connect to that port:

```
$ grep server_url mystorage.conf
      "server_url": "http://localhost:2379",
```

etcd running:

```
$ sudo lsof -i -P | grep etcd | grep 2379
etcd      8574       etcd    6u  IPv4 168765      0t0  TCP deepfryer.redhat.com:2379 (LISTEN)
```

But still error

```
(devel) [~/rhat/commissaire/commissaire-service] 10:40:17  (master)
$ commissaire-storage-service -c mystorage.conf &    
[1] 18461
(devel) [~/rhat/commissaire/commissaire-service] [1] 10:40:19  (master)
$ Traceback (most recent call last):
  File "/home/tbielawa/rhat/commissaire/devel/bin/commissaire-storage-service", line 11, in <module>
    load_entry_point('commissaire-service', 'console_scripts', 'commissaire-storage-service')()
  File "/home/tbielawa/rhat/commissaire/commissaire-service/src/commissaire_service/storage/__init__.py", line 276, in main
    config_file=args.config_file)
  File "/home/tbielawa/rhat/commissaire/commissaire-service/src/commissaire_service/storage/__init__.py", line 53, in __init__
    super().__init__(exchange_name, connection_url, queue_kwargs)
...
  File "/home/tbielawa/rhat/commissaire/devel/lib/python3.5/site-packages/amqp/transport.py", line 120, in connect
    self._connect(self.host, self.port, self.connect_timeout)
  File "/home/tbielawa/rhat/commissaire/devel/lib/python3.5/site-packages/amqp/transport.py", line 161, in _connect
    self.sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

[1]  + exit 1     commissaire-storage-service -c mystorage.conf
```